components/categ-tool-group.vue

```diff
@@ -165,34 +189,147 @@
 
             invalidNameEntered() {
 
-                // Duplicate group names are not allowed
-                return ( this.newToolGroupName in this.toolGroups );
+                // Name cannot be a duplicate of another tool group
+                let invalid = ( this.newToolGroupName in this.toolGroups );
+
+                // With the exception of the current group being edited while in 'modify' mode
+                if ( this.modes.modify === this.currentMode ) {
+                    invalid = invalid && this.currentGroup !== this.newToolGroupName;
+                }
+
+                return invalid;
             },
 
-            readyToAddToolGroup() {
+            readyToAddOrModifyToolGroup() {
+
+                // 1. Minimum ready condition check
+
+                // A. The entered group name must not be blank and there must be
+                // at least one tool selected
+                let ready = ( "" !== this.newToolGroupName &&
+                    this.selectedTools.length > 0 );
+                
+                // B. Short-circuit out if this condition is not met
+                if ( !ready ) {
+                    return false;
+                }
+
+                // 2. Mode-specific additional condition check
+                switch ( this.currentMode ) {
+
+                    case this.modes.create:
+
+                        // No duplicate group names allowed
+                        ready = ( ready &&
+                            !Object.prototype.hasOwnProperty.call(this.toolGroups, this.newToolGroupName) );
+                        break;
+
+                    case this.modes.modify:
+
+                        // No duplicate group names, but skip check for the
+                        // name of the group currently being modified
+                        for ( const groupName in this.toolGroups ) {
+
+                            if ( groupName === this.newToolGroupName &&
+                                groupName !== this.currentGroup ) {
+                                ready = false;
+                                break;
+                            }
+                        }
+                        break;
+                }
+
+                return ready;
+            }
+        },
+
+        watch: {
+
+            columnToCategoryMap: {
+
+                deep: true,
+                handler(p_newColumnToCategoryMap, p_oldColumnToCategoryMap) {
+
+                    // Note: `newValue` will be equal to `oldValue` here
+                    // on nested mutations as long as the object itself
+                    // hasn't been replaced.
+
+                    // 1. Check to see if any columns have been unlinked as assessment tools
+                    const toBeRemoved = [];
+                    for ( const item of this.assessmentToolGroups.items ) {
```

Comment:
There is something weird going on here when the `columnToCategoryMap` have been changed (e.g. by removing a column from a tool group in the annotation window). The problem seems to be that although `this.assessmentToolGroups` clearly has a property `items` with a non-empty array, directly accessing this array via `this.assessmentToolGroups.items` (e.g. `console.log(this.assessmentToolGroups.items)` shows an empty array. 

Take a look here:
![annot_tool_bug](https://user-images.githubusercontent.com/1302022/170143235-3ff4d160-ac4e-4179-9339-cbab25e02ff8.gif)
If you access the variable directly or watch it via vue-dev tools, all looks normal. But obviously you cannot iterate over an empty array so you never get to check if anything has to be removed here.

As I mention in the review, i think the real solution to this is to redesign the way the tool groups are implemented. But we should do this in a later PR, right now we get what we need and this bug is something we can work around.

Reply:
1. Yes, to redesign. One of the reasons I had wanted to take a look at a potential object-oriented reworking is because things are beginning to feel to disparate/unwieldy in the store and in turn too constrictive (a.k.a. 'brittle'). Too many sources of truth. That said, that's not for now. So let's discuss this smaller internals redesign next.
2. I'm going to take a look into this for a little bit so I can better understand what is happening.

Reply:
So, from what I can tell there is some disconnect in how you're viewing the object via the console. Debug messages from source reveal `assessmentToolGroups.items` to be correctly populated.

That said, I have produced a fix for the issue of the table not being updated and though we should readdress how this is done in the redesign we mentioned, we have some nice "magic" at hand.

In the annotation pages `unlinkColumnFromCategory` method - which is called when a column is removed from the `annot-column` component, I have added the following third step to appropriately update the tool groups in the store.

```
                // 3. If this column was a grouped tool, remove it from its group
                if ( this.isToolGrouped(p_event.removedColumn) ) {

                    // A. Remove the tool from its group
                    const groupName = this.getGroupOfTool(p_event.removedColumn);
                    this.$store.dispatch("removeToolFromGroup", {
                        group: groupName,
                        tool: p_event.removedColumn
                    });

                    // B. If its former group is now empty of tools, remove the group itself
                    for ( const groupName in this.toolGroups ) {
                        if ( 0 === this.toolGroups[groupName].length ) {

                            this.$store.dispatch("removeToolGroup", { name: groupName });
                        }
                    }
                }
```

This duplicates behavior from the `categ-tool-group` component itself, but applies the changes we want.  Changes are reflected on the categorization page.

And here's the Vue "magic" part. If every tool is removed from its tab on the annotation page (`annot-columns` component), the tab itself is now removed.